packages/pkh-stacks/src/authmethod.ts

```diff
@@ -0,0 +1,106 @@
+
+export const VERSION = '1'
+export const CHAIN_NAMESPACE = 'stacks'
+
+export const chainIdMap = {
+  mainnet: '1',
+  testnet: '2147483648',
+}
+
+type StacksNetwork = 'mainnet' | 'testnet'
+
+export namespace StacksWebAuth {
+  /**
+   * Get a configured authMethod for an Ethereum account in a web based environment
+   */
+  // eslint-disable-next-line @typescript-eslint/require-await
+  export async function getAuthMethod(
+    stacksProvider: any,
+    account: AccountId
+  ): Promise<AuthMethod> {
+    if (typeof window === 'undefined')
+      throw new Error('Web Auth method requires browser environment')
+    const domain = (window as Window).location.hostname
+
+    return async (opts: AuthMethodOpts): Promise<Cacao> => {
+      opts.domain = domain
+      return createCACAO(opts, stacksProvider, account)
+    }
+  }
+}
+
+export interface SignatureData {
+  signature: string
+  publicKey: string
+}
+
+export type SupportedProvider = {
+  signatureRequest: (payload: string) => Promise<SignatureData>
+}
+
+export function assertSupportedProvider(
+  stacksProvider: any
+): asserts stacksProvider is SupportedProvider {
+  const p = stacksProvider as SupportedProvider
+  if (p.signatureRequest === null || p.signatureRequest === undefined) {
+    throw new Error(`Unsupported provider; provider must implement signatureRequest`)
+  }
+}
+
+async function sign(stacksProvider: any, message: string) {
+  assertSupportedProvider(stacksProvider)
+  const token = createUnsecuredToken({ message })
+  const { signature } = await stacksProvider.signatureRequest(token)
+  return signature
+}
+
+async function createCACAO(
+  opts: AuthMethodOpts,
+  stacksProvider: any,
+  account: AccountId
+): Promise<Cacao> {
+  const now = new Date()
+  const oneWeekLater = new Date(now.getTime() + 7 * 24 * 60 * 60 * 1000)
+
+  const siwStacksMessage = new SiwStacksMessage({
+    domain: opts.domain,
+    address: account.address,
+    statement: opts.statement ?? 'Give this application access to some of your data on Ceramic',
+    uri: opts.uri,
+    version: VERSION,
+    nonce: opts.nonce ?? randomString(10),
+    issuedAt: now.toISOString(),
+    expirationTime: opts.expirationTime ?? oneWeekLater.toISOString(),
+    chainId: account.chainId.reference,
+    resources: opts.resources,
+  })
+
+  const signData = siwStacksMessage.signMessage()
+  const signature = await sign(stacksProvider, signData)
+  siwStacksMessage.signature = signature
+  return Cacao.fromSiwStacksMessage(siwStacksMessage)
+}
+
+/**
+ * Helper function to get an accountId (CAIP10) for an Stacks account by Stacks Provider interface, Connection must implement 'TODO()'
+ */
+// TODO remove ts-ignore
+// @ts-ignore
+export async function getAccountId(stacksProvider: any, address: string): Promise<AccountId> {
+  // TODO hardcoded for now, update once https://github.com/hirosystems/stacks-wallet-web/issues/2991 is resolved
+  const stacksChainId = chainIdMap['mainnet']
```

Comment:
this is the missing part so for the tests, mainnet was used

Reply:
As you mentioned, if there is no way to get the network/chain from the provider, its totally fine to use statically listed chains/networks (as you did above), as long as you include all the mostly commonly used ones 

And with the `getAccountIdByNetwork` that would cover everything, I would remove this 'getAccountId(stacksProvider: any, address: string)" function helper for now until it is supported, otherwise the expect behavior would not be clear